### ISSUE TITLE
fc.yahoo.com on port 443 (https) is not reachable

### ISSUE THREAD

**Issue by sergiomnb** [ranaroussi/yfinance#1924]
### Describe bug

I had used this library some time ago and everything was fine ,I take it up again today and it doesn't work. What must have changed ? did they put a token ? what are these crumbs ? nothing is documented , search and search but nothing to do. So ,like in the good old days ,I edited the file data.py line 163 and changed fc.yahaoo.com with url='https://finance.yahoo.com/quote/AMD',.
Now it works.

### Simple code that reproduces your problem

https://fc.yahoo.com/ on browser

### Debug log

DEBUG    Entering download()
DEBUG     Disabling multithreading because DEBUG logging enabled
DEBUG     Entering history()
DEBUG      Entering _fetch_ticker_tz()
DEBUG       Entering get()
DEBUG        url=https://query2.finance.yahoo.com/v8/finance/chart/ACN
DEBUG        params=frozendict.frozendict({'range': '1d', 'interval': '1d'})
DEBUG        Entering _get_cookie_and_crumb()
DEBUG         cookie_mode = 'basic'
DEBUG         Entering _get_cookie_and_crumb_basic()
ERROR       Failed to get ticker 'ACN' reason: HTTPSConnectionPool(host='fc.yahoo.com', port=443): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fb0d6219a10>: Failed to establish a new connection: [Errno 111] Connection refused'))

### `yfinance` version

0.2.38

**Comment by i418c** [comment]
Check that the domain isn't blocked by a DNS blocker like PiHole.

**Comment by TobiasLaving** [comment]
Also had this issue, editing the file works

To put the pihole discussion to rest I tried with both my wifi and carrier, same result.
Perhaps the fc domain is not working in some parts of the world?
I am in sweden for example.

**Comment by smiron** [comment]
In my case it turned out to be related to unifi Ad Blocking. Disabling this feature does the trick in this case.

**Comment by kwilsonmg** [comment]
Thank you for writing that, @smiron ! That fixed it for me.

**Comment by szotsaki** [comment]
This is definitely an ad blocking issue. The following adlists contain fc.yahoo.com:

```
https://www.github.developerdan.com/hosts/lists/ads-and-tracking-extended.txt:
  fc.yahoo.com
  all.fc.yahoo.com
  de.fc.yahoo.com
  es.fc.yahoo.com
  finance.fc.yahoo.com
  in.fc.yahoo.com
  it.fc.yahoo.com
  jill.fc.yahoo.com
  ap-southeast.jill.fc.yahoo.com
  eu-central.jill.fc.yahoo.com
  us-east.jill.fc.yahoo.com
  us-west.jill.fc.yahoo.com
  log.fc.yahoo.com
  mail.fc.yahoo.com
  news.fc.yahoo.com
  qa-jill.fc.yahoo.com
  stage-jill.fc.yahoo.com
  uk.fc.yahoo.com

https://raw.githubusercontent.com/0Zinc/easylists-for-pihole/master/easyprivacy.txt:
  jill.fc.yahoo.com
```

I think this bug report can be closed since it's not an issue with yfinance.

**Comment by sergiocntr** [comment]
I understand that the issue is not directly related to the yfinance library itself, but I believe it is still important to keep the issue open for the benefit of other users who may encounter the same problem. The issue stems from the fact that the library is trying to check the fc.yahoo.com host, which is often blocked by many AdBlockers. Unfortunately, the log generated by the library is not verbose enough to clearly explain this issue to the user. This lack of transparency makes it difficult to troubleshoot, especially when the host is unreachable.

As a potential workaround, modifying the domain checked by the library to something less commonly blocked, like Apple's ticket domain, could resolve the problem. Since this issue could affect other users who are using AdBlockers or facing similar network restrictions, I believe it’s important to document it so others can find and understand the cause of the problem.

I kindly request that the issue remains open, not necessarily because it’s a bug in yfinance, but because it highlights a real-world scenario that could help others avoid frustration when they encounter the same situation.

Thank you for your understanding!

**Comment by szotsaki** [comment]
Closing this issue won't make it disappear. It will remain here (with all its comments), just its state will transition.

**Comment by sergiocntr** [comment]
I’m closing this issue as I understand that it will remain accessible for others to find. Thank you for the discussion and clarification!

**Comment by ValueRaider** [comment]
Ideally YF would detect that `fc.yahoo.com` is blocked and handle gracefully (nice message), not blow up.

**Comment by kwilsonmg** [comment]
I agree with @ValueRaider that that would be ideal, but it might not be technically possible to tell if it is blocked or just lost internet connectivity. The quick and dirty workaround that just popped into my head would be to ping Google should connectivity to `fc.yahoo.com` fail and, if that worked, suggest to the user that it `fc.yahoo.com` is blocked...but I am not sure how realistic that implementation might be.

**Comment by Andreaux** [comment]
I don't have fc.yahoo.com blocked, yet I can't resolve the name. What am I doing wrong? :)

**Comment by beakaynet** [comment]
It seems to me, that something else is at play here. I've encountered the same problem as OP and editing data.py to point to finance.yahoo.com instead of fc.yahoo.com fixed the issue for me too. However, I can ping fc.yahoo.com just fine. I've also tested several (non-blocking) DNS services and stopping firewalld with the same results.

**Comment by Sodabeh-Taherpanah** [comment]
Why do not try to fix it ? I am going to use import yfinance as yf

**Comment by ValueRaider** [comment]
#1084

**Comment by Andreaux** [comment]
Simply finding fc.yahoo.com in my adblocker's adlists and whitelisting it silently did the trick. Of course that won't manifest right away as DNS already had the reponse cached so it might take some time to ease up, but that literally is what it took to make it work.

**Comment by whosti** [comment]
I had the same problem: fc.yahoo.com was not reachable via port 443. As a solution, I used a different DNS server. My previously configured Pi-hole DNS was blocking the address.
